src/api/app/controllers/source_project_controller.rb

```diff
@@ -19,43 +20,43 @@ def show
       return
     end
 
+    # This implicitly also checks if the user can access the project (for hidden projects).
+    # We have to make sure to initialize the project already at this
+    # point, even we dont need the object in most cases because of that fact.
+    # TODO: Don't implicitly use the finder logic to authorize!
```

Comment:
Tracking this here https://trello.com/c/rKlWGIdR/203-dont-use-projectfindbyname-for-implicit-authorization